valid-form.php

```diff
@@ -0,0 +1,96 @@
+     } else {
+       $valid['course'] = $course;
+     }
+
+     $lecturer = $_POST['lecturer'];
+     if (!$lecturer) {
+       $errors['lecturer'] = 'Името на лектора е задължително поле.';
+     } elseif (strlen($lecturer) > MAX_LENGTH_LECTURER) {
+       $errors['lecturer'] = 'Името на лектора трябва да е по-малко от '.MAX_LENGTH_LECTURER.' символа.';
+     } else {
+       $valid['lecturer'] = $lecturer;
+     }
+
+     $description = $_POST['description'];
+     if (!$description) {
+       $errors['description'] = 'Описанието на предмета е задължително поле.';
+     } elseif (strlen($description) < MIN_LENGTH_DESCRIPTION) {
+       $errors['description'] = 'Описанието на предмета трябва да е по-голямо от '.MIN_LENGTH_DESCRIPTION.' символа.';
+     } else {
+       $valid['description'] = $description;
+     }
+
+     $group = $_POST['group'];
+     if (!$group) {
+       $errors['group'] = 'Трябва да изберете поне една група.';
+     } else {
+       $valid['group'] = $group;
+     }
+
+     $credits = $_POST['credits'];
+     if (!$credits) {
+       $errors['credits'] = 'Кредит е задължително поле.';
+     } elseif (strlen($credits) < MIN_NUMBER_CREDITS) {
+       $errors['credits'] = 'Кредита трябва да е положително число.';
+     } elseif (strlen($credits) > MAX_NUMBER_CREDITS) {
+       $errors['credits'] = 'Кредита може да бъде най-много '.MAX_NUMBER_CREDITS.'.';
+     } else {
+       $valid['credits'] = $credits;
+     }
+
+     $filename = 'logs.txt';
+     if (count($valid) == NUMBER_VALID_INPUTS) {
+      phpAlert("Successful creation!");
+      file_put_contents($filename, "Име на предмет: ", FILE_APPEND | LOCK_EX);
+      file_put_contents($filename, $course, FILE_APPEND | LOCK_EX);
+      file_put_contents($filename, "\n", FILE_APPEND | LOCK_EX);
+      file_put_contents($filename, "Име на преподавател: ", FILE_APPEND | LOCK_EX);
+      file_put_contents($filename, $lecturer, FILE_APPEND | LOCK_EX);
+      file_put_contents($filename, "\n", FILE_APPEND | LOCK_EX);
+      file_put_contents($filename, "Описание на предмет: ", FILE_APPEND | LOCK_EX);
+      file_put_contents($filename, $description, FILE_APPEND | LOCK_EX);
+      file_put_contents($filename, "\n", FILE_APPEND | LOCK_EX);
+      file_put_contents($filename, "Група: ", FILE_APPEND | LOCK_EX);
+      file_put_contents($filename, $group, FILE_APPEND | LOCK_EX);
+      file_put_contents($filename, "\n", FILE_APPEND | LOCK_EX);
+      file_put_contents($filename, "Кредити: ", FILE_APPEND | LOCK_EX);
+      file_put_contents($filename, $credits, FILE_APPEND | LOCK_EX);
+      file_put_contents($filename, "\n", FILE_APPEND | LOCK_EX);
+      file_put_contents($filename, "===================================================", FILE_APPEND | LOCK_EX);
+      file_put_contents($filename, "\n", FILE_APPEND | LOCK_EX);
+     } else {
+      phpAlert("You have error! Look at the file logs.txt and submit again.");
```

Comment:
Nice one! It would be even better if the user is able to see it on the screen itself.